iroh-sync/src/store/fs.rs

```diff
@@ -424,12 +375,12 @@ impl crate::ranger::Store<RecordIdentifier, SignedEntry> for StoreInstance {
     fn put(&mut self, k: RecordIdentifier, v: SignedEntry) -> Result<()> {
         // TODO: propagate error/not insertion?
         if v.verify().is_ok() {
-            let timestamp = v.entry().record().timestamp();
+            let timestamp = k.timestamp();
```

Comment:
this is missing the timestamp check on insert, need to makes sure it is newer than the existing one